src/request.rs

```diff
@@ -365,6 +367,52 @@ impl<'a> RequestBuilder<'a> {
         self.0
     }
 
+    /// Extract the `bewit` query parameter, if any, from the path, and return it.  If the path
+    /// contains no bewit, the RequestBuilder is returned unchanged.  If the path contains an
+    /// invalid bewit, or multiple bewit parameters, the result is an Err.
+    ///
+    /// The path manipulation is tested to correspond to that preformed by the hueniverse/hawk
+    /// implementation-specification
+    pub fn extract_bewit(mut self) -> Result<(Self, Option<Bewit<'a>>)> {
+        const PREFIX: &str = "bewit=";
+
+        if let Some(query_index) = self.0.path.find('?') {
+            let mut bewit_components: Vec<&str> = vec![];
+            let components: Vec<&str> = self.0.path[query_index + 1..]
```

Comment:
You should be able to use https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition here, which would be clearer.

Even then, this allocates more than it needs to, but eh... it's probably fine for now.